jacodb-ets/src/main/kotlin/org/jacodb/ets/dto/Signatures.kt

```diff
@@ -18,18 +18,59 @@ package org.jacodb.ets.dto
 
 import kotlinx.serialization.Serializable
 
+@Serializable
+data class FileSignatureDto(
+    val projectName: String,
+    val fileName: String,
+) {
+    override fun toString(): String {
+        return "@$projectName/$fileName"
+    }
+}
+
+@Serializable
+data class NamespaceSignatureDto(
+    val name: String,
+    val declaringFile: FileSignatureDto? = null,
+    val declaringNamespace: NamespaceSignatureDto? = null,
+) {
+    init {
+        require(!(declaringFile != null && declaringNamespace != null)) {
```

Comment:
Definitely not XOR, since both declaring file and namespace might be missing (in DTO). We do have a non-nullable `declaringFile` in our model, but with a great chance we are going to loosen this requirement and also have both file/namespace nullable. The reason for this is that `Class<->File` relation is not as strong as `Class<->Method`, so we might encounter situations when we do not have a corresponding File for a Class.